astroquery/cadc/core.py

```diff
@@ -275,6 +276,162 @@ def get_collections(self):
+            of str if returning urls).
+        """
+
+        filenames = self.get_images_async(coordinates, radius, collection,
+                                          get_url_list, show_progress)
+
+        if get_url_list:
+            return filenames
+
+        images = []
+
+        for fn in filenames:
+            try:
+                images.append(fn.get_fits())
+            except HTTPError as err:
+                # Catch HTTPError if user is unauthorized to access file
+                logger.debug(
+                    "{} - Problem retrieving the file: {}".
+                    format(str(err), str(err.url)))
+                pass
+
+        return images
+
+    def get_images_async(self, coordinates, radius, collection=None,
+                         get_url_list=False, show_progress=False):
+        """
+        A coordinate-based query function that returns a list of
+        context managers with cutouts around the passed in coordinates.
+
+        Parameters
+        ----------
+        coordinates : str or `astropy.coordinates`.
+            Coordinates around which to query.
+        radius : str or `astropy.units.Quantity`
+            The radius of the cone search AND cutout area.
+        collection: str, optional
+            Name of the CADC collection to query.
+        get_url_list: bool, optional
+            If true, returns the list of data urls rather than
+            the list of context managers. Default is `False`.
+        show_progress: bool, optional
+            Whether to display a progress bar if the file is downloaded
+            from a remote server.  Default is `False`.
+
+        Returns
+        -------
+        list : A list of context-managers that yield readable file-like objects
+        """
+        request_payload = self._args_to_payload(coordinates=coordinates,
+                                                radius=radius,
+                                                collection=collection,
+                                                data_product_type='image')
+        query_result = self.exec_sync(request_payload['query'])
+        images_urls = self.get_image_list(query_result, coordinates, radius)
+
+        if get_url_list:
+            return images_urls
+
+        return [commons.FileContainer(url, encoding='binary',
+                                      show_progress=show_progress)
+                for url in images_urls]
+
+    def get_image_list(self, query_result, coordinates, radius):
+        """
+        Function to map the results of a CADC query into URLs to
+        corresponding data and cutouts that can be later downloaded.
+
+        The function uses the IVOA DataLink Service
+        (http://www.ivoa.net/documents/DataLink/) implemented at the CADC.
+        It works directly with the results produced by Cadc.query_region and
+        Cadc.query_name but in principle it can work with other query
```

Comment:
these can probably be backticked to link to the appropriate modules, e.g. `Cadc.query_region`.  I'm not sure whether the full path is needed in scope here...

Reply:
You are right! Full path is not needed.